core/renderers/zelos/info.js

```diff
@@ -202,6 +215,44 @@ Blockly.zelos.RenderInfo.prototype.addAlignmentPadding_ = function(row,
   }
 };
 
+/**
+ * Adjust the x position of fields to bump all non-label fields in the first row
+ * past the notch position.
+ * @protected
+ */
+Blockly.zelos.RenderInfo.prototype.adjustXPosition_ = function() {
+  if (!this.topRow.hasPreviousConnection) {
+    return;
+  }
+  var firstInputRow = null;
+  for (var i = 0, row; (row = this.rows[i]); i++) {
+    if (!firstInputRow && Blockly.blockRendering.Types.isInputRow(row)) {
+      firstInputRow = row;
+      var minXPos = this.constants_.NOTCH_OFFSET_LEFT +
+          this.constants_.NOTCH_WIDTH;
+      var xCursor = row.xPos;
+      var prevSpacer = null;
+      for (var j = 0, elem; (elem = row.elements[j]); j++) {
+        if (Blockly.blockRendering.Types.isSpacer(elem)) {
+          prevSpacer = elem;
+        }
+        if (Blockly.blockRendering.Types.isField(elem) ||
+            Blockly.blockRendering.Types.isInput(elem)) {
+          if (xCursor < minXPos &&
+              !(Blockly.blockRendering.Types.isField(elem) &&
```

Comment:
This is saying that label fields should be not shunted but inputs and non-label fields should be moved, right?

Reply:
Correct.